server/src/main/java/com/cloud/agent/manager/allocator/impl/FirstFitAllocator.java

```diff
@@ -318,10 +321,8 @@ protected List<Host> allocateTo(DeploymentPlan plan, ServiceOffering offering, V
 
             //find number of guest VMs occupying capacity on this host.
             if (_capacityMgr.checkIfHostReachMaxGuestLimit(host)) {
-                if (s_logger.isDebugEnabled()) {
-                    s_logger.debug("Host name: " + host.getName() + ", hostId: " + host.getId() +
-                        " already has max Running VMs(count includes system VMs), skipping this and trying other available hosts");
-                }
+                s_logger.debug(String.format("Adding host [%s] to the avoid set because this host already has the max number of running (user and/or system) VMs.",
```

Comment:
I restrain myself to not ask for extra guards. Please do not remove them when there. even when upgraded to log4j3 we'll have to remove the `String.format` calls to prevent the parameterexpansions.

Reply:
Fixed :)